lib/pageflow/file_types.rb

```diff
@@ -7,7 +7,9 @@ def initialize(page_types)
     end
 
     def each(&block)
-      @page_types.map(&:file_types).flatten.uniq(&:model).each(&block)
+      first_level_file_types = @page_types.map(&:file_types).flatten.uniq(&:model)
+      lower_level_file_types = search_for_nested_file_types(first_level_file_types)
+      (first_level_file_types + lower_level_file_types).flatten.uniq(&:model).each(&block)
```

Comment:
No need to call `flatten` and `unique` twice. Test for nested case still missing.